src/kafka-extensions/registry.ts

```diff
@@ -121,16 +122,28 @@ function sortByNameAscending(a: ClusterProvider, b: ClusterProvider): -1 | 0 | 1
     return 0;
 }
 
-function intializeIfNeeded() {
+function initializeIfNeeded() {
     if (providers.size === 0) {
-        providers = collectClusterProviderDefinitions(vscode.extensions.all);
+        refreshClusterProviderDefinitions();
     }
 }
 
 export interface ClusterProviderDefinition {
     id: string;
     name?: string;
 }
+export function refreshClusterProviderDefinitions() {
+    // Collect cluster provider extensions
+    const oldClusterProviderIds = Array.from(providers.keys());
+    providers = collectClusterProviderDefinitions(vscode.extensions.all);
+
+    // Disconnect all kafka client linked to a cluster provider id coming from an installed/uninstalled extension
+    const newClusterProviderIds = Array.from(providers.keys());
+    const allClusterProviderIds = [...oldClusterProviderIds, ...newClusterProviderIds]
+        .filter(clusterProviderId => clusterProviderId !== defaultClusterProviderId);
+    const toDispose = [...new Set(allClusterProviderIds)];

```

Comment:
no, you should only dispose old is that are not in the new newClusterProviderIds.
Something like:
```
idsToDispose = oldClusterProviderIds.filter( id => !newClusterProviderIds.includes(id) )
```

Reply:
Thanks for your suggestion, I tried on my side an dit works great. Fixed.